src/components/safe-apps/AppFrame/index.tsx

```diff
@@ -106,11 +106,13 @@ const AppFrame = ({ appUrl, allowedFeaturesList }: AppFrameProps): ReactElement
       origin: document.location.origin,
     }),
     onGetSafeInfo: useGetSafeInfo(),
-    onGetSafeBalances: (currency) =>
-      getBalances(chainId, safeAddress, currency, {
+    onGetSafeBalances: (currency) => {
+      const isDefaultTokenlistSupported = chain && hasFeature(chain, FEATURES.DEFAULT_TOKENLIST)
+      return getBalances(chainId, safeAddress, currency, {
         exclude_spam: true,
-        trusted: TOKEN_LISTS.TRUSTED === tokenlist,
-      }),
+        trusted: isDefaultTokenlistSupported && TOKEN_LISTS.TRUSTED === tokenlist,
+      })
+    },
```

Comment:
Just looked back over the codebase and we're using the same logic in `useLoadBalances`.

What do you think about using its return value or abstracting that logic it into a simpler hook to use in both places?

Reply:
I also thought about simply returning the balances from our `balanceSlice`.
That would also be much faster and they get updated / polled regularly.

But I wasn't sure if there is a specific reason we fetch the balances fresh from our service here.